src/WebJobs.Extensions.CosmosDB/Config/CosmosDBExtensionConfigProvider.cs

```diff
@@ -53,12 +53,13 @@ public void Initialize(ExtensionConfigContext context)
                 throw new ArgumentNullException("context");
             }
 
-            // Apply ValidateConnection to all on this rule. 
+            // Apply ValidateConnection to all on this rule.
             var rule = context.AddBindingRule<CosmosDBAttribute>();
             rule.AddValidator(ValidateConnection);
             rule.BindToCollector<DocumentOpenType>(typeof(CosmosDBCollectorBuilder<>), this);
 
             rule.BindToInput<CosmosClient>(new CosmosDBClientBuilder(this));
+            rule.BindToInput<ParameterBindingData>((attr) => CreateParameterBindingData(attr));
```

Comment:
For Blob extension, we had changes in [equivalent file](https://github.com/surgupta-msft/azure-sdk-for-net/blob/60bc0fc45ab345c83aa4eebec17a27adb1b1b892/sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/src/Config/BlobsExtensionConfigProvider.cs#L111) for supporting collection. Do we not need similar changes in case of cosmos extension or may be I am missing something?

Reply:
We do not, cosmos doesn't have an "IsBatched" concept so we don't need to support ParameterBindingData collection - this is all handled in the worker

Reply:
Nice, good to know.